src/initialize_dae.jl

```diff
@@ -152,20 +146,32 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
 
         nlequation! = @closure (out, u, p) -> begin
             update_coefficients!(M, u, p, t)
-            #M * (u-u0)/dt - f(u,p,t)
+            # f(u,p,t) + M * (u0 - u)/dt
```

Comment:
why the sign flip?

Reply:
the sign flip made the Jacobian a little cleaner IMO. I could revert if you prefer.